### PR TITLE
SPL Tokens Support

### PR DESCRIPTION
**Pull request by omkarshanbhag**
This PR adds support for parsing SPL transfers. Specifically it parses for the following transfer instructions across botth the `TOKEN_PROGRAM` and `TOKEN_2022_PROGRAM`: 
- `Transfer`
- `TransferChecked`
- `TransferCheckedWithFee`

It then converts any such parsed instructions into an instance of the following type to be output by the parser 

```
#[derive(Debug, Clone, PartialEq)]
pub struct SplTransfer {
    pub from: String,
    pub to: String,
    pub amount: String,
    pub owner: String,
    pub signers: Vec<String>, // This is an empty array if ths is not a multisig account with multiple signers 
    pub token_mint: Option<String>, 
    pub decimals: Option<String>,
    pub fee: Option<String>,
}
```

This code has been tested with unittests on payloads that include transaction with SPL token transfer instructions of various types (calling both token programs, all three instructions, single signer multi signer etc)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,8 @@ edition = "2021"
 
 [dependencies]
 log = "0.4.22"
-solana-sdk = "2.0.3"
-solana-program = "2.0.3"
+solana-sdk = "2.1.4"
```
Comment (andrewkmin) [inline Cargo.toml:8]: are these version bumps necessary? if required by `spl-token`, can we use `6.0.0`?

Comment (omkarshanbhag) [reply Cargo.toml:8]: bumps not necessary! I've also removed the other dependency and done the parsing explicitly.

Comment (omkarshanbhag) [reply Cargo.toml:8]: TODO -- remove dependency bump

File: src/solana/parser.rs [hunk 2]
```diff
@@ -421,6 +425,55 @@ fn read_compact_u16(tx_body_remainder: &[u8]) -> Result<(usize, &[u8]), Box<dyn
     }
 }
 
+// SplInstructionData represents parsed instruction data of an instruction to the solana token program or the token 2022 program
+enum SplInstructionData {
```
Comment (omkarshanbhag) [inline src/solana/parser.rs:429]: TODO -- add references to spl token and spl token 2022 code

**Review by andrewkmin — APPROVED** [review]
looking good overall! 🚀 

let's update the README as well with this new batch of functionality :)

**Comment by omkarshanbhag** [comment]
README updated!! Merging now.

**Pull request merged**